libbeat/publisher/client.go

```diff
@@ -106,13 +115,9 @@ func (c *client) annotateEvent(event common.MapStr) {
 	}
 }
 
-func (c *client) getClient(opts []ClientOption) (context, eventPublisher) {
-	var ctx context
-	for _, opt := range opts {
-		ctx = opt(ctx)
-	}
-
-	if ctx.sync {
+func (c *client) getClient(opts []ClientOption) (Context, eventPublisher) {
+	ctx := makeContext(opts)
+	if ctx.Sync {
```

Comment:
This code makes it look like, Async is the default (which I think it isn't). The code is correct, just my first impression.